In [5]:
import pandas as pd

datasets = ['amazon', 'celeba', 'gaussian', 'moons']
# Path to your CSV file
data = datasets[1]
csv_file_path = f'./history/metrics_{data}.csv'

# Read the CSV file into a DataFrame
df = pd.read_csv(csv_file_path)
# df = df.iloc[:-1]
# Print the DataFrame
print(df)


   seed data_name amazon_cat      mode  best_train_loss  best_train_jaccard  \
0     1    celeba       toys  implicit         3.449098           63.072667   
1     2    celeba       toys  implicit         3.451227           56.130333   
2     3    celeba       toys  implicit         3.450038           59.472333   
3     4    celeba       toys  implicit         3.448667           65.510000   
4     5    celeba       toys  implicit         3.449505           62.497667   

   best_val_jaccard  best_test_jaccard  epoch_time         time  \
0         52.943333          50.686667   43.804581  2541.858834   
1         52.533333          50.776667   44.498330  1522.920430   
2         51.430000          50.593333   44.158453  1988.278793   
3         52.433333          50.856667   44.074054  2631.399937   
4         53.596667          52.570000   43.954295  2349.570950   

   memory_used_MB                                         train_loss  \
0     3339.746094  [3.456784500357548, 3.454813922

In [49]:
import pandas as pd

datasets = ['amazon', 'celeba', 'gaussian', 'moons']
# Path to your CSV file
data = datasets[3]
csv_file_path = f'./history/baseline_metrics_{data}.csv'

# Read the CSV file into a DataFrame
df = pd.read_csv(csv_file_path)

df = df[df['mode']=='ind']
# Get the column names
columns = list(df.columns)

# Move the second-to-last column to the 9th position (index 8)
col_to_move = columns[-2]
columns.insert(8, columns.pop(columns.index(col_to_move)))

# Reorder the DataFrame
df = df[columns]

print(df.head())

   seed data_name amazon_cat mode  best_train_loss  best_train_jaccard  \
5     1     moons       toys  ind         8.793417           55.940580   
6     2     moons       toys  ind         8.856157           55.197096   
7     3     moons       toys  ind         9.048162           55.999458   
8     4     moons       toys  ind         8.447986           55.500764   
9     5     moons       toys  ind         8.029468           55.364859   

   best_val_jaccard  best_test_jaccard       time              epoch_time  \
5         56.722111          57.035697  18.317459  0 days 00:00:00.457155   
6         56.575572          55.793595  16.907937  0 days 00:00:00.446780   
7         57.562196          56.792146  17.298204  0 days 00:00:00.458624   
8         56.602234          56.663263  17.977662  0 days 00:00:00.445770   
9         56.463665          56.086284  19.418471  0 days 00:00:00.453035   

                                          train_loss  \
5  [12.748395562171936, 11.719006180

In [6]:
# Select all columns except the first four

columns_to_analyze = df.columns[4:-4]

# Group by 'mode'
grouped = df.groupby('amazon_cat')

# Calculate mean and standard deviation for each group
mean_df = grouped[columns_to_analyze].mean()
std_df = grouped[columns_to_analyze].std()

# Create a new DataFrame to store the formatted mean±std values
formatted_df = mean_df.copy()

# Format the mean and standard deviation to two decimal places and combine them
for col in columns_to_analyze:
    formatted_df[col] = mean_df[col].map('{:.2f}'.format) + '±' + std_df[col].map('{:.2f}'.format)

# Print the formatted DataFrame
print(f'dataset name:  {data}')
print("Formatted Mean±Std values:")
print(formatted_df)

dataset name:  celeba
Formatted Mean±Std values:
           best_train_loss best_train_jaccard best_val_jaccard  \
amazon_cat                                                       
toys             3.45±0.00         61.34±3.62       52.59±0.79   

           best_test_jaccard  epoch_time            time  
amazon_cat                                                
toys              51.10±0.83  44.10±0.26  2206.81±455.08  


dataset name:  celeba
best_train_loss            3.45±0.00
best_train_jaccard        60.35±1.82
best_val_jaccard          53.61±0.31
best_test_jaccard         51.40±0.73
epoch_time                44.30±0.47
time                  2220.82±258.07

dataset name:  gaussian
best_train_loss         13.86±0.00
best_train_jaccard      89.69±0.08
best_val_jaccard        90.35±0.08
best_test_jaccard       90.99±0.03
epoch_time               0.58±0.01
time                    36.28±0.43

dataset name:  moons
best_train_loss         13.86±0.00
best_train_jaccard      57.71±0.14
best_val_jaccard        58.92±0.08
best_test_jaccard       58.52±0.06
time                    33.96±0.26


dataset name:  amazon
Formatted Mean±Std values:
           best_train_loss best_train_jaccard best_val_jaccard  \
amazon_cat                                                       
apparel          6.31±0.00         68.41±0.00       68.84±0.00   
bath             5.26±0.00         76.83±1.24       77.49±0.79   
bedding          5.32±0.00         76.10±0.00       74.90±0.00   
carseats         4.61±0.00         20.95±0.10       22.94±0.17   
diaper           5.75±0.00         82.49±0.00       81.84±0.00   
feeding          6.40±0.00         80.17±0.00       79.72±0.00   
furniture        4.34±0.00         19.60±0.10       21.31±0.01   
gear             5.42±0.00         60.53±0.00       60.05±0.00   
health           5.07±0.00         75.12±0.00       72.65±0.00   
media            6.23±0.00         63.08±0.00       59.72±0.00   
safety           4.39±0.00         26.79±0.00       29.48±0.00   
toys             5.60±0.00         68.60±0.06       67.81±0.46   

           best_test_jaccard epoch_time         time  
amazon_cat                                                           
apparel           66.78±0.00  0.60±0.03   48.38±0.28    
bath              76.74±1.21  0.51±0.02  66.91±12.73    
bedding           74.41±0.00  0.59±0.02   56.41±0.37   
carseats          19.71±0.39  0.29±0.03   32.72±0.61  
diaper            81.45±0.00  0.67±0.02   81.18±0.96   
feeding           79.86±0.00  0.88±0.03   82.18±1.09    
furniture         17.46±0.39  0.25±0.02   29.69±1.44     
gear              59.65±0.00  0.58±0.03   43.14±0.69     
health            71.57±0.00  0.49±0.03   95.08±0.90    
media             59.51±0.00  0.38±0.01   88.32±0.67   
safety            28.52±0.00  0.27±0.03   40.12±1.48     
toys              67.12±0.37  0.43±0.03   64.06±0.86  